### DISCUSSION TITLE
How to get a persistent multi-column layout

### DISCUSSION THREAD

**Discussion by Firedrake** [sile-typesetter/sile#2153] [Q&A]
I am very new to SILE, and I want a multi-column layout.

(Actually I ultimately want a mixed layout, where an article starts with a full-page-width title, then proceeds with two-column under that. But first things first.)

The book tells me how to set a two-column pagetemplate. The examples show me how to do it with frametricks. Both of these work. But the instant the text flows onto the next page, it's back to single column.

Is there a way to say "stick with two columns on all succeeding pages", or even better "until the end of this environment"?

**Comment by Omikhleia** [comment]
Greetings,
- On having a frame layout that sticks on all succeeding pages, consider using "masters" (higher-level abstraction) rather than the (lower-level) "pagetemplate" --> as illustrated in this similar question #1919. It's touchy, but it works (I've done several booklets with "sections" in two-column mode, though eventually I had to abstract most of the complexity into a Lua package).[^1]
- On mixed layouts (i.e. switching back and forth between full-width and two-column on the same page = as in a  two-column layout but sometimes going back to single column on a page (for section title etc.))... I'm afraid there's nothing yet out of the box.[^2]
- The experimental frametricks package may help there indeed, _maybe_. It also may call for more troubles, though, it is not a tamed beast!

[^1]: If you want to dabble into arcane code, see e.g. my [teidict.sile](https://github.com/Omikhleia/teidict.sile) module: the class there might have some interesting bits of code (though it also does other very specialized things).
[^2]: I once did something that was _kind of_ working in simple cases, but it failed in many other cases (footnotes were also an issue in such a layout); I went on working on other topics... Surely I'll resurrect my own attempt at one point, but I can't say -- it took many years for LaTeX to have a working "muticol" package, and there are good reasons to that, it seems.
